ably/realtime_channel_spec_test.go

```diff
@@ -0,0 +1,265 @@
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		channel = c.Channels.Get("test")
+
+		_, err = channel.Attach()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ablytest.Instantly.Recv(t, nil, out, t.Fatalf) // Consume ATTACH
+
+		stateChanges = make(chan ably.State, 10)
+		channel.On(stateChanges)
+
+		return
+	}
+
+	t.Run("RTL13a: when ATTACHED, successful reattach", func(t *testing.T) {
+		t.Parallel()
+
+		in, out, _, channel, stateChanges, afterCalls := setup(t)
+
+		// Get the channel to ATTACHED.
+
+		in <- &proto.ProtocolMessage{
+			Action:  proto.ActionAttached,
+			Channel: channel.Name,
+		}
+
+		var change ably.State
+
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.StateChanAttached, change.State; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, change)
+		}
+
+		errInfo := proto.ErrorInfo{
+			StatusCode: 500,
+			Code:       50500,
+			Message:    "fake error",
+		}
+
+		in <- &proto.ProtocolMessage{
+			Action:  proto.ActionDetached,
+			Channel: channel.Name,
+			Error:   &errInfo,
+		}
+
+		// Expect a transition to ATTACHING with the error.
+
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.StateChanAttaching, change.State; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, change)
+		}
+		if got := fmt.Sprint(change.Err); !strings.Contains(got, errInfo.Message) {
+			t.Fatalf("expected %+v; got %v (error: %+v)", errInfo, got, change.Err)
+		}
+
+		var msg *proto.ProtocolMessage
+		ablytest.Instantly.Recv(t, &msg, out, t.Fatalf)
+		if expected, got := proto.ActionAttach, msg.Action; expected != got {
+			t.Fatalf("expected %v; got %v (message: %+v)", expected, got, msg)
+		}
+
+		// TODO: Test attach failure too, which requires RTL4f.
+
+		in <- &proto.ProtocolMessage{
+			Action:  proto.ActionAttached,
+			Channel: channel.Name,
+		}
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.StateChanAttached, change.State; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, change)
+		}
+		if change.Err != nil {
+			t.Fatal(change.Err)
+		}
+
+		// Expect the retry loop to be finished.
+		ablytest.Instantly.NoRecv(t, nil, afterCalls, t.Fatalf)
+		ablytest.Instantly.NoRecv(t, nil, stateChanges, t.Fatalf)
+	})
+
+	t.Run("RTL13b: when ATTACHING", func(t *testing.T) {
+		t.Parallel()
+
+		in, out, _, channel, stateChanges, afterCalls := setup(t)
+
+		errInfo := proto.ErrorInfo{
+			StatusCode: 500,
+			Code:       50500,
+			Message:    "fake error",
+		}
+
+		in <- &proto.ProtocolMessage{
+			Action:  proto.ActionDetached,
+			Channel: channel.Name,
+			Error:   &errInfo,
+		}
+
+		// Expect a state change with the error.
+
+		var change ably.State
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.StateChanDetached, change.State; expected != got {
```

Comment:
@tcard , this should be `ChannelStateSusended` as per [spec](https://docs.ably.io/client-lib-development-guide/features/#RTL13b)
I was looking into RTL13 because it's a bit related to channel attach

Reply:
Just to add, if we change it to as per spec, we will lose one of the next state for transitioner, from
`ATTACHING` to `DETACHED`, so we will have to update transitioner, it's working for now